main.cpp

```diff
@@ -56,64 +56,81 @@ Create a branch named Part1
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(float v, const char* w) : video(v), waveForm(w) {}  //1
+    float video; //2
+    std::string waveForm; //3
 };
 
-struct <#structName1#>                                //4
+struct CompareCalc                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if( a->video < b->video ) return a;
+        if( a->video > b->video ) return b;
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float x { 0 }, y { 0 };
+    float nValue(float* outPut)      //12
     {
-        
+        x = *outPut;
+
+        if(x >= y)
+        {
+            x += x * y;
+        }
+        else
+        {
+            y -= x * y;
+        }
+
+        return x *= y;
```

Comment:
The body of this function should be essentially identical to the static function below

- [ ] revise function body